modules/generating-icsp-object-scoped-to-a-registry.adoc

```diff
@@ -0,0 +1,64 @@
+[id="generating-icsp-object-scoped-to-a-registry_{context}"]
+= Widening the mirror registry scope of the ImageContentSourcePolicy object
+
+[role="_abstract"]
+Use the `--icsp-scope=registry` flag with the `oc adm catalog` command to widen the mirror registry scope in the ICSP file. This minimizes the number of times the cluster nodes are rebooted when a registry entry is updated.
+
+The `oc adm catalog mirror` command generates an ImageContentSourcePolicy (ICSP) object that maps the original container image repository to a new location where it is mirrored, typically inside a disconnected environment. When a new or modified ICSP is applied to a cluster, it is converted to a config file for CRI-O and placed onto each node. The process of placing the config file on a node includes rebooting that node. Having a widely scoped ICSP reduces the number of times the ICSP needs to change in the future and thus reduces the number of times a cluster must reboot its nodes.
+
+.Prerequisites
+
+* Install the {product-title} CLI `oc`.
+* Log in as a user with `cluster-admin` privileges.
+* Configure a local image catalog repository with restricted OLM for use in your {product-title} disconnected cluster.
+
+.Procedure
+
+To widen the mirror registry scope in the ICSP file, do the following:
+
+. Run the following command, specifying values for <local_registery> and <pull_secret_file>:
++
+[source,terminal]
+----
+$ oc adm catalog mirror <local_registery>/olm/redhat-operators:v1 <local_registery> -a <pull_secret_file> --icsp-scope=registry --manifests-only=true
```

Comment:
Wouldn't disconnected environments need to mirror the actual images? Should we remove the `--manifests-only=true` flag?

Reply:
Given that you ask users to interact with the mapping files, you'll need to add a `oc image mirror` call to mirror the contents of the registry.

Reply:
This is a typo i think. We don't need to cover this `--manifests-only=true` parameter in the context of this procedure. It's covered in https://docs.openshift.com/container-platform/4.7/operators/admin/olm-restricted-networks.html#olm-mirror-catalog_olm-restricted-networks